aws/resource_aws_cognito_identity_provider.go

```diff
@@ -43,11 +54,24 @@ func resourceAwsCognitoIdentityProvider() *schema.Resource {
 			"provider_name": {
 				Type:     schema.TypeString,
 				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 32),
+					validation.StringMatch(regexp.MustCompile(`^[^_][\p{L}\p{M}\p{S}\p{N}\p{P}][^_]+$`), "see https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateIdentityProvider.html#API_CreateIdentityProvider_RequestSyntax"),
+				),
 			},
 
 			"provider_type": {
 				Type:     schema.TypeString,
 				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					cognitoidentityprovider.IdentityProviderTypeTypeSaml,
+					cognitoidentityprovider.IdentityProviderTypeTypeFacebook,
+					cognitoidentityprovider.IdentityProviderTypeTypeGoogle,
+					cognitoidentityprovider.IdentityProviderTypeTypeLoginWithAmazon,
+					cognitoidentityprovider.IdentityProviderTypeTypeOidc,
```

Comment:
Since this was introduced, there is an additional value, according to the [SDK constants](https://docs.aws.amazon.com/sdk-for-go/api/service/cognitoidentityprovider/#pkg-constants):

```suggestion
					cognitoidentityprovider.IdentityProviderTypeTypeOidc,
					cognitoidentityprovider.IdentityProviderTypeTypeSignInWithApple,
```